pkg/src/index.js

```diff
@@ -436,6 +443,58 @@ export async function publint({ pkgDir, vfs, level, strict, _packedFiles }) {
     }
   }
 
+  /**
+   * @param {Record<string, string> | string} repositoryField
+   */
+  async function checkRepositoryField(repositoryField) {
+    if (typeof repositoryField === 'string') {
+      if (!isRepositoryUrl(repositoryField)) {
```

Comment:
I'm not sure if checking it's a github url only is a good idea. Some users may use gitlab or bitbucket to host for example, and that shouldn't raise a warning either.

Maybe here's some things we can lint based on the [docs](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#repository):
- If it's a string and looks like a URL, we should warn to use an object instead
- If it's an object with `type: 'git'` and the url contains `github.com` or `gitlab.com`, make sure the url starts with `git+` and ends with `.git`

I tested these mostly by running `npm publish --dry-run` in a test project and see it's warnings.

Reply:
1. The string format is a bit vague, I think we can just throw a `suggestion` if it's a string that suggest the user to use object notation for `repository`.
2. After some testing, I believe [this regex](https://www.debuggex.com/r/cleXSxwdY3n9BaMV) is able to address a lot `git` URL

Reply:
1. Using a string for `repository` is valid though and I think it comes down to preference. We should recommend the object form if we can know if the string format is invalid.
2. From what I can tell, it's valid for someone to put `https://my-git-host.com/project` and npm doesn't seem to warn about this. It's only for `github.com` and `gitlab.com` that npm knows you need a `git+` and `.git`. I'm ok if we want to recommend always adding the `git+` and `.git` as a start, maybe that's a good default for external hosts too, but something to watch for in the future if it may not be the case.

Reply:
1. Good point, I'll push a `suggestion` if it's a string.
2. After digging around, it seems that actually there's a [specification for Git URLs](https://git-scm.com/docs/git-clone#_git_urls). We might want to implement this specification instead. What do you think?

Reply:
1. To clarify, I don't think we should push a `suggestion` if it's _any_ kind of string as it's not wrong to use `"repository": "github:user/repo"` for example. Only strings that aren't documented to be valid in the [npm docs](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#repository) and they should be a `warning` (not suggestion since other package managers like pnpm doesn't autofix 🤔)
2. Interesting. I'm ok if we try to enforce this format always then since it looks like best practice to follow a git url spec.

Reply:
Resolved all of these